### ISSUE TITLE
A compileOnly dependency is used in the Kotlin/Native target

### ISSUE THREAD

**Issue by eygraber** [Kotlin/kotlinx-atomicfu#397]
After updating to 0.23.2 the following gets printed when building:

```
A compileOnly dependency is used in the Kotlin/Native target 'iosArm64':
Compilation: main

Dependencies:
org.jetbrains.kotlinx:atomicfu:0.23.2

Such dependencies are not applicable for Kotlin/Native, consider changing the dependency type to 'implementation' or 'api'.
To disable this warning, set the kotlin.native.ignoreIncorrectDependencies=true project property
```

I am using atomicfu as a plugin (`id("kotlinx-atomicfu")`) and in some cases as a dependency (`implementation(libs.kotlinx.atomicfu)`).

**Comment by lppedd** [comment]
See explanation in https://github.com/Kotlin/kotlinx-atomicfu/pull/395

TL;DR: this will go away in Kotlin 2.0

**Comment by mvicsokolova** [comment]
Thank you, yes, this warning will be resolved in Kotlin 2.0. 

`kotlinx-atomicfu` adds a `compileOnly` dependency to the library for the common sourceSet (this caused the issue), and `implementation` dependency to the native sourceSet.

**Comment by eygraber** [comment]
This is still happening in Kotlin 2.0 RC1. Is that expected?

**Comment by eygraber** [comment]
Still seeing this in Kotlin 2.0 stable. Does atomicfu need to be updated to Kotlin 2.0 for this to go away?

**Comment by mvicsokolova** [comment]
The fix of this warning in the compiler did not get to `2.0.0` release. 
It will be available since `2.0.20-Beta1`.
I've checked the build with the dev version `2.0.20-dev-4770` -- there are no warnings in the build log.

Here is the YT issue: [KT-64109](https://youtrack.jetbrains.com/issue/KT-64109/Using-compileOnly-runtimeOnly-dependencies-in-K-N-related-configurations-leads-to-odd-behaviour)
